helpers/path.go

```diff
@@ -75,6 +75,35 @@ func (filepathBridge) Separator() string {
 
 var fpb filepathBridge
 
+// segmentReplacer replaces some URI-reserved characters in a path segments.
+var segmentReplacer = strings.NewReplacer("/", "-", "#", "-")
+
+// MakeSegment returns a copy of string s that is appropriate for a path
+// segment.  MakeSegment is similar to MakePath but disallows the '/' and
+// '#' characters because of their reserved meaning in URIs.
+func (p *PathSpec) MakeSegment(s string) string {
+	s = p.MakePathSanitized(strings.Trim(segmentReplacer.Replace(s), "- "))
+
+	var pos int
+	var last byte
+	b := make([]byte, len(s))
+
+	for i := 0; i < len(s); i++ {
```

Comment:
If I read this correctly this removes double-slashes. I assume that come from "My // Taxonomy" cases. Which is fine as long as this isn't used in the "general case" (which would break some URLs and make people grumpy). But you should add a test variant for it below.

Reply:
Yes, we would replace both slashes with `-` using `segmentReplacer` (other interesting characters like a space and `?` are replaced by `MakePathSanitized` used above), and this loop removes consecutive dashes.

So, `My // Taxonomy` would become `my-taxonomy`.

Reply:
Fixed.